include/unifex/finally.hpp

```diff
@@ -700,39 +701,41 @@ namespace unifex
         : source_(static_cast<SourceSender2&&>(source))
         , completion_(static_cast<CompletionSender2&&>(completion)) {}
 
-      // TODO: Also constrain this method to check that the CompletionSender
+      // TODO: Also constrain these methods to check that the CompletionSender
       // is connectable to any of the instantiations of done/value/error_receiver
       // that could be created for each of the results that SourceSender might
       // complete with. For now we just check done_receiver as an approximation.
+
       template <
           typename Receiver,
           typename CPO,
           typename S,
           std::enable_if_t<
               std::conjunction_v<
                 std::is_same<CPO, tag_t<connect>>,
-                std::is_same<S, sender>,
+                std::is_same<std::remove_cvref_t<S>, sender>,
+                std::negation<std::is_lvalue_reference<S>>,
```

Comment:
Was it necessary to limit to rvalue-reference only here so that the subsequent `is_connectable<SourceSender/CompletionSender` checks are using the correct value-category?

ie. they'd need to be checking `is_connectable<const SourceSender&, ...>` and also `is_copy_constructible<CompletionSender>` and `is_connectable<CompletionSender, ...>`.

Do we still need to add this other const-lvalue connect overload?

Reply:
I cannot specify an rvalue overload of tag_invoke. it always creates a forwarding overload of tag_invoke when using &&. I just decided to use the same forwarding overload to handle both rvalue and const lvalue.

Reply:
`std::is_lvalue_reference<S>` will still evaluate to true if `S` is a const-ref.
So this constraint would seem to only match rvalue-references as it stands.

Consider changing to `std::negation<std::is_same<S, sender&>>` if you want it to match `const sender&` and `sender&&`.
But if you do that then you also need to change the `is_connectable` check below to use the corresponding value category for the SourceSender check.